docs/advanced/witness.md

```diff
@@ -0,0 +1,82 @@
+
+Harvester now supports the witness node involved cluster. You can enable the witness node when joining the cluster.
+For information about the role management, please refer to [role management](../TBD.md).
+
+
+:::note
+
+The current limitation is that the witness node can only be enabled when joining the cluster. Also, the cluster can only have one witness node.
+
+:::
+
+## Create Harvester cluster with witness node
+
+When you create a new cluster, you can choose the witness role when joining the cluster.
+For example, you create a cluster with `harvester-node-0`, `harvester-node-1` and `harvester-node-2`.
+
+Choose the `harvester-node-1` as the witness node when joining the cluster. After the cluster is created, you would see the `harvester-node-1` only has the etcd capability as follows:
+
+```
+NAME↑               STATUS   ROLE                         VERSION               PODS     CPU      MEM    %CPU    %MEM    CPU/A    MEM/A AGE        
+harvester-node-0    Ready    control-plane,etcd,master    v1.27.10+rke2r1         70    1095    10143      10      63    10000    15976 4d13h      
+harvester-node-1    Ready    etcd                         v1.27.10+rke2r1          7     258     2258       2      14    10000    15976 4d13h      
+harvester-node-2    Ready    control-plane,etcd,master    v1.27.10+rke2r1         36     840     6905       8      43    10000    15976 4d13h      
+```
+
+You can see the `harvester-node-1` use less of resource to running as expection.
+
+## Workloads on the witness node
+
+The witness node only runs the must workload like etcd. Now the following list is the supported workloads on the witness node.
+- **harvester-node-manager**
+- **cloud-controller-manager**
+- **etcd**
+- **kube-proxy**
+- **rke2-canal**
+- **rke2-multus**
+
+## Upgrade the cluster with witness node
+
+When you upgrade the cluster with a witness node, you can upgrade the cluster as usual. But if you have a degraded volume, the upgrade might fail. You can check the following limitations for the volume robustness.
+
+## Limitations
+
+There were some limitations for the cluster with witness.
+
+- Volume robustness: The witness node does not store any data. It means we will miss one replica when the cluster has the witness node due to the Longhorn default replica number (3). The upgrade might failed by the degraded volume.
```

Comment:
```suggestion
- Volume robustness: The witness node does not store any volume data. It means we will miss one replica when the cluster has the witness node due to the Longhorn default replica number (3). The upgrade might failed by the degraded volume.
```

It also happens when there are only 3 nodes in the cluster.

Reply:
updated!